### PR TITLE
bloomrpc: init at 1.5.3

### PR DESCRIPTION
**Pull request by zoedsoupe**
###### Motivation for this change
Add a GUI Client for gRPC Services! It's so similar with `insomnia` and `postman` but for
gRPC protocol.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/web/bloomrpc/default.nix [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    openssl
+    stdenv.cc.cc
+  ];
+in stdenv.mkDerivation rec {
+  pname = "bloomrpc";
+  version = "1.5.2";
+
+  src = fetchurl {
+    url =
+      "https://github.com/uw-labs/bloomrpc/releases/download/${version}/bloomrpc_${version}_amd64.deb";
```
Comment (SuperSandro2000) [inline pkgs/development/web/bloomrpc/default.nix:23]:
```suggestion
    url = "https://github.com/uw-labs/bloomrpc/releases/download/${version}/bloomrpc_${version}_amd64.deb";
```

Can we build from source instead?

File: pkgs/development/web/bloomrpc/default.nix [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    sha256 = "9da17bcd3ec579309661658849652e2bf8e0db6961716e94af299ee41f0fd778";
+  };
+
+  nativeBuildInputs =
+    [ autoPatchelfHook dpkg makeWrapper gobject-introspection wrapGAppsHook ];
```
Comment (SuperSandro2000) [inline pkgs/development/web/bloomrpc/default.nix:28]:
```suggestion
  nativeBuildInputs = [ autoPatchelfHook dpkg makeWrapper gobject-introspection wrapGAppsHook ];
```

File: pkgs/development/web/bloomrpc/default.nix [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+  installPhase = ''
+    mkdir -p $out/share/bloomrpc $out/lib $out/bin
+
+    mv usr/share/* $out/share/
+    mv opt/BloomRPC/* $out/share/bloomrpc/
+    mv $out/share/bloomrpc/*.so $out/lib/
+
+    ln -s $out/share/bloomrpc/bloomrpc $out/bin/bloomrpc
+    sed -i 's|\/opt\/BloomRPC|'$out'/bin|g' $out/share/applications/bloomrpc.desktop
```
Comment (SuperSandro2000) [inline pkgs/development/web/bloomrpc/default.nix:77]: Please use substituteInPlace.

File: pkgs/development/web/bloomrpc/default.nix [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/uw-labs/bloomrpc";
+    description = "GUI Client for GRPC Services";
+    license = licenses.lgpl3Only;
+    platforms = [ "x86_64-linux" ];
+    maintainers = with maintainers; [ Mdsp9070 ];
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/development/web/bloomrpc/default.nix:92]:
```suggestion
  };
}
```

File: pkgs/development/web/bloomrpc/default.nix [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+let
+  runtimeLibs = lib.makeLibraryPath [
+    curl
+    glibc
+    libudev0-shim
+    nghttp2
+    openssl
+    stdenv.cc.cc
+  ];
+in stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/web/bloomrpc/default.nix:17]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/development/web/bloomrpc/default.nix [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    mv usr/share/* $out/share/
+    mv opt/BloomRPC/* $out/share/bloomrpc/
+    mv $out/share/bloomrpc/*.so $out/lib/
+
+    ln -s $out/share/bloomrpc/bloomrpc $out/bin/bloomrpc
+    sed -i 's|\/opt\/BloomRPC|'$out'/bin|g' $out/share/applications/bloomrpc.desktop
+  '';
+
+  preFixup = ''
+    wrapProgram "$out/bin/bloomrpc" --prefix LD_LIBRARY_PATH : ${runtimeLibs}
```
Comment (SuperSandro2000) [inline pkgs/development/web/bloomrpc/default.nix:81]:
```suggestion
    wrapProgram "$out/bin/bloomrpc" --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [
    curl
    glibc
    libudev0-shim
    nghttp2
    openssl
    stdenv.cc.cc
  ]}
```

**Comment by lukegb** [comment]
Thanks for the contribution! :) PTAL at the outstanding review comments.

File: pkgs/development/web/bloomrpc/default.nix [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    install -m 444 -D ${appimageContents}/${pname}.desktop $out/share/applications/${pname}.desktop
+    install -m 444 -D ${appimageContents}/${pname}.png \
+      $out/share/icons/hicolor/512x512/apps/${pname}.png
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun' 'Exec=${pname}'
+  '';
+
+  meta = with lib; {
+    description = "Inspired by Postman and GraphQL Playground
+BloomRPC aims to provide the simplest and most efficient developer experience for exploring and querying your GRPC services.";
```
Comment (SuperSandro2000) [inline pkgs/development/web/bloomrpc/default.nix:34]: This is to long for description. use longDescription instead.

File: pkgs/development/web/bloomrpc/default.nix [hunk 8]
```diff
@@ -0,0 +1,40 @@
+{ lib, fetchurl, appimageTools, pkgs }:
+
+let
+  pname = "bloomrpc";
+  version = "1.5.3";
+  name = "${pname}-${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/web/bloomrpc/default.nix:6]:
```suggestion
```
all appimage tools now also understand pname.

File: pkgs/development/web/bloomrpc/default.nix (hunk 7 above)
Comment (matdsoupe) [reply pkgs/development/web/bloomrpc/default.nix]: did it

File: pkgs/development/web/bloomrpc/default.nix (hunk 8 above)
Comment (matdsoupe) [reply pkgs/development/web/bloomrpc/default.nix]: did it

File: pkgs/development/web/bloomrpc/default.nix [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    ln -s $out/bin/${name} $out/bin/${pname}
+    install -m 444 -D ${appimageContents}/${pname}.desktop $out/share/applications/${pname}.desktop
+    install -m 444 -D ${appimageContents}/${pname}.png \
+      $out/share/icons/hicolor/512x512/apps/${pname}.png
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun' 'Exec=${pname}'
+  '';
+
+  meta = with lib; {
+    longDescription = "Inspired by Postman and GraphQL Playground
```
Comment (SuperSandro2000) [inline pkgs/development/web/bloomrpc/default.nix:33]: There should always be description. longDescription is an addition to it. Also longDescription should use multiline strings, not a normal string with newlines.

File: pkgs/development/web/bloomrpc/default.nix (hunk 8 above)
Comment (SuperSandro2000) [reply pkgs/development/web/bloomrpc/default.nix:6]: Thanks! I think we no longer need this line.

File: pkgs/development/web/bloomrpc/default.nix [hunk 10]
```diff
@@ -0,0 +1,40 @@
+{ lib, fetchurl, appimageTools, pkgs }:
```
Comment (SuperSandro2000) [inline pkgs/development/web/bloomrpc/default.nix:1]:
```suggestion
{ lib, fetchurl, appimageTools }:
```

I think the construct in extraPkgs does not need this input.

File: pkgs/development/web/bloomrpc/default.nix [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+    install -m 444 -D ${appimageContents}/${pname}.png \
+      $out/share/icons/hicolor/512x512/apps/${pname}.png
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun' 'Exec=${pname}'
+  '';
+
+  meta = with lib; {
+    description = "GUI Client for GRPC Services";
+    longDescription = ''
+    Inspired by Postman and GraphQL Playground BloomRPC aims to provide the simplest and most efficient developer experience for exploring and querying your GRPC services.
```
Comment (SuperSandro2000) [inline pkgs/development/web/bloomrpc/default.nix:34]:
```suggestion
      Inspired by Postman and GraphQL Playground BloomRPC aims to provide the simplest and most efficient developer experience for exploring and querying your GRPC services.
```

**Comment by SuperSandro2000** [comment]
Same here with the eval error.

**Pull request merged**